### DISCUSSION TITLE
Local file repository

### DISCUSSION THREAD

**Discussion by Pinion999** [FlamxGames/godot-ai-assistant-hub#5] [Ideas]
Would it be possible to add a button to set a local folder or files for the AI to read and use the contents?

**Comment by FlamxGames** [comment]
For the assistant resources?
Can you elaborate what benefit are you trying to get?

**Reply by Pinion999** [reply]
I'm not sure if it's called embedded documents. In the chat application GPT4All, you can use LocalDocs, which is powered by Nomic Embedding models for the AI to use your documents as a knowledge base for it's responses.
To load in all scripts in the current project, to train the AI specifically to the current project and give responses that encompass and take into account the rest of the project.
To load in the latest version of the Godot manual, to get better responses.
To load in a repository of scripts or templates, to further train the AI.

**Reply by FlamxGames** [reply]
Oh, I understand what you mean now. My understanding is that, to do that we would also need a Vector DB, not sure how GPT4All does it, I assume it may have some Vector DB internally, I believe Nomic library is what provides the logic to convert the docs into embeddings, but at the end there should be something else to work as the Vector DB.

Yes, I think it would be great to have this functionality. But it also sounds like it could bring a lot of complexity. Rather than getting into the business of repeating what GPT4All already does, maybe we can integrate the plugin with GPT4All, I just checked and they seem to also have an API similar to Ollama. Not sure if that uses the embedded docs though.

Right now I can't get into trying to do this myself, but if you (or anyone reading this) want to give it a try, the video I created about how to extend the plugin should help to get started: [https://www.youtube.com/watch?v=A03gNLgZ6eA](https://www.youtube.com/watch?v=A03gNLgZ6eA), just replace what I did with ChatGPT API with GPT4All API.

**Reply by Linshiqi** [reply]
Hi @FlamxGames, thank you for creating this awesome project! I discovered it while planning to build something similar myself - in fact, I found your solution before writing any code. It covers almost all the features I need, but I'd like to suggest adding one more capability: enabling the AI assistant to read entire project scenes and GDScript files as context. This would allow the AI to use your documentation as a knowledge base for its responses, similar to what's been discussed in previous threads.

You might want to check out this implementation: https://github.com/mkgiga/godot-ai-helper. They've successfully integrated the ability to add scene and script files to the AI's context. I think incorporating similar functionality here would be valuable. From my understanding, this wouldn't require model retraining or vector databases - simply providing the file contents to the AI should suffice. What are your thoughts on this approach?

On a personal note, I'm primarily a C++ developer exploring game development. While I'm new to this field, I'd be happy to contribute to the project in any way I can. Please feel free to reach out if you think I could help with specific tasks.

**Reply by FlamxGames** [reply]
@Linshiqi thank you for your kind words.

Sending the context of the files in the chat itself could work, I'm not sure how it would handle multiple files and how much would that increase the response time. I agree that, in general, extending this plugin to be aware of your project would be the next logical step to improve this tool. However I'm not planning to invest on it right now since I need to invest all my time on my current game, and to be honest I'm not using this plugin at all.

However, don't let that stop you, I do review, test, and merge pull requests, recently two alternatives to Ollama were added in that way, if you feel like getting into extending this, I encourage you to give it a try.
